### PR TITLE
HLM-6172 updating operation to match api changes

### PR DESCRIPTION
**Pull request by siddhant-nawale-egov**

### CODE REVIEW

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/utils/processHierarchyAndData.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ export const processHierarchyAndData = (hierarchy, allData) => {
     Object.values(hierarchicalData).forEach((country) => {
       if (country.children[null]) {
         country.data = country.children[null].data;
-        delete country.children[null];
+        country.children[null] = undefined;
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/utils/processHierarchyAndData.js:68]: It was origanally null and you asked to make it this way

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/utils/geojsonValidations.js [hunk 2]
```diff
@@ -112,9 +112,9 @@ export const geojsonPropetiesValidation = (data, schemaData, name, t) => {
   };
   const validateGeojson = ajv.compile(schema);
   const valid = validateGeojson(data);
-  let errors = {};
+  const errors = {};
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/utils/geojsonValidations.js:115]: its a string enum

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/utils/excelValidations.js [hunk 3]
```diff
@@ -37,127 +38,146 @@ export const excelValidations = (data, schemaData, t) => {
...
+    }
+  });
+
+  return {
+    valid: !hasDataErrors,
+    message: errorMessages ? [...new Set(Object.values(errorMessages))] : [],
+    errors,
+    missingColumnsList,
+  };
+};
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/utils/excelValidations.js:166]: its localised

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/MicroplanPreviewHelperCompoenents.js [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+export const EditResourceData = ({ previewData, selectedRow, resources, tempResourceChanges, setTempResourceChanges, data, t }) => {
+  const conmmonColumnData = useMemo(() => {
+    const index = previewData?.[0]?.indexOf(commonColumn);
+    if (index === -1) return;
+    return previewData?.[selectedRow]?.[index];
+  }, [previewData]);
+
+  const valueChangeHandler = (item, value) => {
+    if (!conmmonColumnData) return;
+    if (isNaN(value) || (!isFinite(value) && value !== "")) return;
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/MicroplanPreviewHelperCompoenents.js:340]: i want coercion

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/MicroplanPreviewHelperCompoenents.js [hunk 5]
```diff
@@ -0,0 +1,434 @@
...
+              </tr>
+            </thead>
+            <tbody>
+              {previewData.slice(1).map((rowData, rowIndex) => {
+                const rowDataList = Object.values(previewData[0]).map((header, cellIndex) => (
+                  <td
+                    className={`${selectedRow && selectedRow - 1 === rowIndex ? "selected-row" : ""}`}
+                    key={cellIndex}
+                    style={{
+                      ...(rowData[cellIndex] || rowData[cellIndex] === 0 ? (!isNaN(rowData[cellIndex]) ? { textAlign: "end" } : {}) : {}),
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/MicroplanPreviewHelperCompoenents.js:220]: i want coercion

File: micro-ui/web/micro-ui-internals/example/src/ComponentRegistry.js [hunk 6]
```diff
@@ -1,11 +0,0 @@
-class Registry {
```
Comment (nipunarora-eGov) [inline micro-ui/web/micro-ui-internals/example/src/ComponentRegistry.js:1]: @siddhant-nawale-egov this file is not required?

File: micro-ui/web/micro-ui-internals/example/src/index.js [hunk 7]
```diff
@@ -52,22 +51,21 @@ const initTokens = (stateCode) => {
...
   initMicroplanningComponents();
 
   const moduleReducers = (initData) => initData;
 
-
   const stateCode = window?.globalConfigs?.getConfig("STATE_LEVEL_TENANT_ID") || "pb";
   initTokens(stateCode);
 
-  ReactDOM.render(<DigitUI stateCode={stateCode} enabledModules={enabledModules}       defaultLanding="employee"  moduleReducers={moduleReducers} />, document.getElementById("root"));
+  ReactDOM.render(
```
Comment (nipunarora-eGov) [inline micro-ui/web/micro-ui-internals/example/src/index.js:65]: formatting here @siddhant-nawale-egov

**Review by nipunarora-eGov — APPROVED** [review]

**Pull request merged**